ui/js/directives/perf/compare.js

```diff
@@ -21,25 +20,33 @@ treeherder.directive(
                     return !matchText || item.toLowerCase().indexOf(matchText.toLowerCase()) > (-1);
                 }
                 scope.filterTest = function(item) {
-                    return filter(item, scope.testFilter);
+                    return _.any(scope.filter.split(' '), function(matchText) {
+                        return filter(item, matchText);
+                    });
                 };
                 scope.filterPlatform = function(result) {
-                    return filter(result.name, scope.platformFilter) &&
-                        (!scope.showOnlyImportant || result.isMeaningful) &&
-                        (!scope.showOnlyConfident || result.isConfident) &&
-                        (scope.showUnreliablePlatforms || !_.contains(
+                    return _.any(scope.filter.split(' '), function(matchText) {
+                        return filter(result.name, matchText);
+                    });
```

Comment:
We shouldn't filter the platform seperately from the test. For each result, we should probably group the test + platform into a single string, then run all the filter terms against that. If all of them match, we're good.